py3status/modules/mpris.py

```diff
@@ -0,0 +1,378 @@
+        """
+        players_paused = []
+        players_stopped = []
+        players = []
+        players_prioritized = []
+
+        _dbus = self._dbus.get('org.freedesktop.DBus')
+        for player in _dbus.ListNames():
+            if SERVICE_BUS in player:
+                player = re.sub(r'%s' % SERVICE_BUS_REGEX, '', player)
+                players.append(player)
+
+        if self.player_priority is None:
+            players_prioritized = players
+        else:
+            player_priority = self.player_priority.split(',')
+
+            for player_name in player_priority:
+                if player_name in players:
+                    players.remove(player_name)
+                    players_prioritized.append(player_name)
+            if '*' in player_priority:
+                players_prioritized += players
+
+        for player_name in players_prioritized:
+            player_state = self._get_state(player_name)
+
+            if player_state == 'Playing':
+                return player_name
+            elif player_state == 'Paused':
+                players_paused.append(player_name)
+            else:
+                players_stopped.append(player_name)
+
+        if players_paused:
+            return players_paused[0]
+        if players_stopped:
+            return players_stopped[0]
+
+        return 'None'
+
+    def _get_text(self, i3s_config):
+        """
+        Get the current metadatas
+        """
+        is_stream = False
+        player = UNKNOWN
+        state = UNKNOWN
+        artist = UNKNOWN
+        album = UNKNOWN
+        length = UNKNOWN_TIME
+        time = UNKNOWN_TIME
+        title = UNKNOWN
+
+        try:
+            player = self._player.Identity
+            (color, state) = self._get_state_format(i3s_config)
+
+            _time_ms = self._player.Position
+            time = _get_time_str(_time_ms)
+
+            metadata = self._player.Metadata
+            if len(metadata) > 0:
+                is_stream = 'file://' not in metadata.get('xesam:url')
+                title = metadata.get('xesam:title') or UNKNOWN
+                album = metadata.get('xesam:album') or UNKNOWN
+
+                if metadata.get('xesam:artist') is not None:
+                    artist = metadata.get('xesam:artist')[0]
+                else:
+                    # we assume here that we playing a video and these types of
+                    # media we handle just like streams
+                    is_stream = True
+
+                _length_ms = metadata.get('mpris:length')
+                if _length_ms is not None:
+                    length = _get_time_str(_length_ms)
+            else:
+                # use stream format if no metadata is available
+                is_stream = True
+        except Exception:
+            color = i3s_config['color_bad']
+
+        if is_stream:
+            # delete the file extension
+            title = re.sub(r'\....$', '', title)
```

Comment:
Your regex `r'\....$'` only catches 3 letter extensions `r'\.[^.]+*$'` might be better.  But is hiding the extension important?  I'm not sure.  Do all streams have extensions? what if there is a `.` in a 'proper' title